src/games/behavmixed.imp

```diff
@@ -368,8 +321,6 @@ T MixedBehaviorProfile<T>::GetLiapValue(bool p_definedOnly) const
 
   T x, result = ((T) 0), avg, sum;
 
-  // HACK: force it to recompute data.  FIX THIS.
-  m_cacheValid = false;
   ComputeSolutionData();
 
   for (int i = 1; i <= m_support.GetGame()->NumPlayers(); i++) {
```

Comment:
You might just want to crib the implementation of this from `liap_solve`.  We have an open question about how to deal with normalisation of the profiles and/or negative probabilities.

Reply:
Understood on the open question, but separately shouldn't we only have one implementation? I can crib it for now..

Reply:
Ah I now see the comment:

// Liapunov function proper - should be replaced with call to profile once
// the penalty is removed from that implementation.